Tools/jit/_targets.py

```diff
@@ -44,6 +44,7 @@ class _Target(typing.Generic[_S, _R]):
     stable: bool = False
     debug: bool = False
     verbose: bool = False
+    known_symbols: dict[str | None, int] = dataclasses.field(default_factory=dict)
```

Comment:
Why do we allow `None` keys? I'm noticing this in all of the type hints where it's used.

Reply:
The dictionary stores `hole.symbol` as key and this is defined as "str | None" hence I need to follow this.

Reply:
Hm, but it should never *be* `None` in the `dict`, right? I think we can remove `None` from the type hint here (and in `process_relocations`, and the two writer functions) if we just assert that the symbol isn't `None` before inserting it into the dict.

Not a huge deal, but I think it better represents what we're actually working with here.

Reply:
Done, it should work now.